test/testpg/pgstresstest.go

```diff
@@ -0,0 +1,188 @@
+
+	"log"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const metaCRD = `
+kind: "postgresql"
+apiVersion: "acid.zalan.do/v1"
+
+metadata:
+  name: "acid-pg-test"
+  labels:
+    team: acid
+
+spec:
+  teamId: "acid"
+  postgresql:
+    version: "14"
+  numberOfInstances: 3
+  volume:
+    size: 500Mi
+
+  users:
+    pg-test: [superuser,createdb]
+    
+  databases:
+    pg-db: pg-test
+    
+  resources:
+    requests:
+      cpu: 300m
+      memory: 300Mi
+    limits:
+      cpu: 300m
+      memory: 300Mi
+`
+
+//you can also copy this file to your project and run it or run pod on sealos cloud.
+//and also you can write a go test to run this Stress Test.
+//what you need is to specify a kubeconfig path and your namespace
+//example
+//func main() {
+//	PostgresStressTest("~/.kube/config", "ns-b47ef502")
+//}
+
+func PostgresStressTest(configpath, userNamespace string) {
+	err := RunPostgresTest(configpath, userNamespace)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func RunPostgresTest(config, userNamespace string) error {
+	var (
+		gvk *schema.GroupVersionKind
+		dr  dynamic.ResourceInterface
+		err error
+	)
+	obj := &unstructured.Unstructured{}
+	_, gvk, err = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).Decode([]byte(metaCRD), nil, obj)
+	if err != nil {
+		return err
+	}
+	dr, err = GetGVRdyClient(gvk, config, userNamespace)
+	if err != nil {
+		return err
+	}
+	for {
+		ctx := context.Background()
+		createObj, err := PostgresClusterRandomCreate(ctx, dr, obj)
+		if err != nil {
+			return err
+		}
+		time.Sleep(time.Minute * 1)
+		for _, obj := range createObj {
+			err := PostgresGet(ctx, dr, obj)
+			if err != nil {
+				return err
+			}
+		}
+		time.Sleep(time.Minute * 1)
+		for _, obj := range createObj {
+			PostgresDelete(ctx, dr, obj)
+		}
+		time.Sleep(time.Minute * 15)
+	}
+}
+
+// PostgresClusterRandomCreate Create random PostgresCluster.
+func PostgresClusterRandomCreate(ctx context.Context, dr dynamic.ResourceInterface, obj *unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	var err error
+	rand.Seed(time.Now().UnixNano())
+	n := rand.Intn(5)
+	objSlice := []*unstructured.Unstructured{}
+	//Create
+	for i := 0; i < n; i++ {
+		objTmp := obj.DeepCopy()
+		objTmp.SetName(objTmp.GetName() + strconv.Itoa(i))
+		_, err = dr.Create(ctx, objTmp, metav1.CreateOptions{})
+		if err != nil {
+			log.Printf("Create %v err : %v", objTmp.GetName(), err)
+			continue
+		}
+		log.Printf("Create %v success", objTmp.GetName())
+		objSlice = append(objSlice, objTmp)
+	}
+	return objSlice, err
+}
+
+// PostgresGet Get PostgresCluster.
+func PostgresGet(ctx context.Context, dr dynamic.ResourceInterface, obj *unstructured.Unstructured) error {
+	objGET, err := dr.Get(ctx, obj.GetName(), metav1.GetOptions{})
+	if err != nil {
+		log.Printf("select resource ERROR: %v\n", err)
+		return err
+	}
+	fmt.Println(objGET.Object["status"])
```

Comment:
Just check the status, not show it.

Reply:
eee,what about record it in log? If we go back to check log ,know cluster status is make sense.